src/Storage.js

```diff
@@ -0,0 +1,69 @@
+import redis from 'redis'
+
+class Storage {
+
+  // options.host options.port
+  constructor(options){
+    const opts = options || {}
+    if (!redis.createClient) throw new Error('Storage: Redis is not available in this enviroment, use functionality which does not require persistent storage')
+    this.client = redis.createClient(opts)
+    this.client.on("error", err => {console.log("Error " + err)})
+    this.challengeKeyPrefix = opts.challengeKeyPrefix || `challenge`
+    this.responseKeyPrefix = opts.responseKeyPrefix || `response`
+    this.challengeKey = pairId => (`${this.challengeKeyPrefix}:${pairId}`)
+    this.responseKey = pairId => (`${this.responseKeyPrefix}:${pairId}`)
+  }
+
+  writeChallenge(pairId, challenge) {
+  return this.set(this.challengeKey(pairId), challenge)
+  }
+
+  readChallenge(pairId) {
+    return this.get(this.challengeKey(pairId))
+  }
+
+  deleteChallenge(pairId) {
+    return this.del(this.challengeKey(pairId))
+  }
+
+  writeResponse(pairId, response) {
```

Comment:
Storing responses is for the desktop browser flow, similar to the messaging server. Once we we make the messaging server available, developers could deploy there own messaging server to use along with this server (or maybe want it on a single server). Mainly we can't use the uPort messaging server to these types of responses since anyone controlling the messaging server could could use any/all of these response to login at a given service as a given user.